src/typings/unix.interface.ts

```diff
@@ -11,3 +11,18 @@ export const isUnixError = (val: any): val is UnixError => {
     'code' in val &&
     'message' in val;
 };
+
+export interface NodeError {
+  syscall: string;
+  path: string;
+  code: string;
+  message: string;
+  errno: number;
+}
+
+export const isNodeError = (val: any): val is NodeError => {
```

Comment:
This can just be
```suggestion
export const isNodeError = (val: any): val is NodeJS.ErrnoException => {
```

Reply:
I did that and ended up with lint messages in the code. `NodeError` has `code` while `NodeJS.ErrnoException` has `code?`. It could be a string or undefined. `code` is used in places where undefined won't work. So, this is really looking for a `ErrnoException` with populated fields.